src/Elastic.Apm/Filters/SpanStackTraceCapturingFilter.cs

```diff
@@ -0,0 +1,54 @@
+// Licensed to Elasticsearch B.V under
+// one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+using System.Diagnostics;
+using Elastic.Apm.Api;
+using Elastic.Apm.Helpers;
+using Elastic.Apm.Logging;
+using Elastic.Apm.Model;
+using Elastic.Apm.ServerInfo;
+
+namespace Elastic.Apm.Filters
+{
+	/// <summary>
+	/// Stack trace capturing itself happens on the application thread (in order to get the real stack trace).
+	/// This filter turns <see cref="Span.RawStackTrace" /> (which is a plain .NET System.Diagnostics.StackTrace instance) into
+	/// <see cref="Span.StackTrace" /> (which is the representation of the intake API stacktrace model).
+	/// This can be done on a non-application thread right before the span gets sent to the APM Server.
+	/// </summary>
+	internal class SpanStackTraceCapturingFilter
+	{
+		private readonly IApmServerInfo _apmServerInfo;
+		private readonly IApmLogger _logger;
+
+		public SpanStackTraceCapturingFilter(IApmLogger logger, IApmServerInfo apmServerInfo) =>
+			(_logger, _apmServerInfo) = (logger, apmServerInfo);
+
+		public ISpan Filter(ISpan iSpan)
```

Comment:
minor nitpick: use `span` for the interface argument and something else for the conversion to `Span` e.g. `concreteSpan`, `capturedSpan`, `realSpan`, etc.